### PR TITLE
auth: Move method checking to Router

### PR DESCRIPTION
**Pull request by cmouse**
### Short description
This change moves the HTTP verb checking into YaHTTP router. 

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by cmouse** [comment]
Upstream PR https://github.com/cmouse/yahttp/pull/30

File: pdns/ws-auth.cc
Comment (cmouse) [reply pdns/ws-auth.cc:1847]: @Habbie I'm wondering if the API should indeed fail with 404 when a zone is not found? Now it just returns empty 200 OK result.

Comment (cmouse) [reply pdns/ws-auth.cc:1847]: ping, this is last issue left

Comment (Habbie) [reply pdns/ws-auth.cc:1847]: empty 200 matches the [documentation](https://doc.powerdns.com/authoritative/http-api/zone.html):

> When set to the name of a zone, only this zone is returned. If no zone with that name exists, the response is an empty array. 

so we can keep that

**Comment by Habbie** [comment]
This has >50 clang-tidy warnings

**Review by Habbie — COMMENTED** [review]
Looks good to me! Will need a review from Otto as well. There's a conflict in ws-auth.cc currently. If Otto agrees, we can finally merge this (after that conflict)

File: pdns/recursordist/ws-recursor.cc
Comment (omoerbeek) [inline pdns/recursordist/ws-recursor.cc:486]:
```suggestion
static void apiServerRPZStats(HttpRequest* /* req */, HttpResponse* resp)
```

Comment (omoerbeek) [reply pdns/recursordist/ws-recursor.cc:486]: Same for `prometheusMetrics`

File: pdns/ws-auth.cc
Comment (omoerbeek) [inline pdns/ws-auth.cc:932]: Not terribly happy with this macro, but I do not see a way around it..

File: pdns/recursordist/ws-recursor.cc
Comment (omoerbeek) [inline pdns/recursordist/ws-recursor.cc:89]:
```suggestion
static void apiServerConfigACLGET(const std::string& aclType, HttpRequest* /* req */, HttpResponse* resp)
```

**Review by omoerbeek — CHANGES_REQUESTED** [review]
A round of review done, will also do some basic tests

File: pdns/ws-auth.cc
Comment (Habbie) [reply pdns/ws-auth.cc:932]: yes, I had the same feeling

File: pdns/ws-auth.cc
Comment (zeha) [inline pdns/ws-auth.cc:944]: what are we gaining from this throw?

Comment (omoerbeek) [reply pdns/ws-auth.cc:944]: The handling of `HttpInternalServerErrorException` (which is derived from in `HttpException`) in `WebServer::handleRequest` is different, though the only difference I see is in what's being logged.

**Review by omoerbeek — COMMENTED** [review]
Did a (basic) review with some nits. I also did some tests for rec and they seem to work as expected. As the regression tests also run fine, I am happy (especially as rec is in that start of a release cycle). Once the nits and the open question are answered, I'm fine with this.

File: pdns/recursordist/ws-recursor.cc
Comment (omoerbeek) [inline pdns/recursordist/ws-recursor.cc:392]: In the case of rec, this can  be a function returning a `DNSName`. The callers then become:
```
auto zonename = findZoneById(req);
```

**Review by omoerbeek — CHANGES_REQUESTED** [review]

**Review by omoerbeek — APPROVED** [review]

**Pull request merged**
